pkg/services/ngalert/remote/alertmanager.go

```diff
@@ -255,45 +255,60 @@ func (am *Alertmanager) CompareAndSendConfiguration(ctx context.Context, config
 	if err := am.autogenFn(ctx, am.log, am.orgID, &c.AlertmanagerConfig, true); err != nil {
 		return err
 	}
-	decrypted, err := am.decryptConfiguration(ctx, c)
+	rawDecrypted, configHash, err := am.decryptConfiguration(ctx, c)
 	if err != nil {
 		return err
 	}
 
 	// Send the configuration only if we need to.
-	if !am.shouldSendConfig(ctx, decrypted) {
+	if !am.shouldSendConfig(ctx, configHash) {
 		return nil
 	}
 
-	isDefault, err := am.isDefaultConfiguration(decrypted)
+	isDefault, err := am.isDefaultConfiguration(configHash)
 	if err != nil {
 		return err
 	}
 
-	return am.sendConfiguration(ctx, decrypted, config.ConfigurationHash, config.CreatedAt, isDefault)
-}
-
-func (am *Alertmanager) isDefaultConfiguration(cfg *apimodels.PostableUserConfig) (bool, error) {
-	rawCfg, err := json.Marshal(cfg)
+	decrypted, err := notifier.Load(rawDecrypted)
 	if err != nil {
-		return false, err
+		return err
 	}
 
-	configHash := fmt.Sprintf("%x", md5.Sum(rawCfg))
+	return am.sendConfiguration(ctx, decrypted, config.ConfigurationHash, config.CreatedAt, isDefault)
+}
 
-	return configHash == am.defaultConfigHash, nil
+func (am *Alertmanager) isDefaultConfiguration(configHash [16]byte) (bool, error) {
+	return fmt.Sprintf("%x", configHash) == am.defaultConfigHash, nil
 }
 
-func (am *Alertmanager) decryptConfiguration(ctx context.Context, cfg *apimodels.PostableUserConfig) (*apimodels.PostableUserConfig, error) {
+// decryptConfiguration decrypts the configuration in-place and returns the decrypted configuration alongside its hash.
+// Should not be used outside of this package and the specific use case of decrypting the configuration before sending
+// it to the remote Alertmanager.
+func (am *Alertmanager) decryptConfiguration(ctx context.Context, cfg *apimodels.PostableUserConfig) ([]byte, [16]byte, error) {
 	fn := func(payload []byte) ([]byte, error) {
 		return am.decrypt(ctx, payload)
 	}
-	decrypted, err := cfg.Decrypt(fn)
+
+	// Iterate through receivers and decrypt secure settings.
+	// It's not necessary to be careful about not modifying the original, as it's used only in a specific context where
+	// the config is read from json and then immediately sent to the remote Alertmanager.
+	for _, rcv := range cfg.AlertmanagerConfig.Receivers {
```

Comment:
This should eventually be replaced with the newer/more robust types and encryption/decryption logic for receivers used in `receiver_svc`, but for now this will work.